File: LabPoo/src/main/polimorfismo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Estudiante = exports.Profesor = void 0;
exports.presentarPersona = presentarPersona;
const encapsulamiento_1 = require("./encapsulamiento");
class Profesor extends encapsulamiento_1.Persona {
    constructor(nombre, edad, sexo, materia) {
        super(nombre, edad, sexo);
        this.materia = materia;
    }
    getMateria() {
        return this.materia;
    }
    setMateria(materia) {
        this.materia = materia;
    }
    presentar() {
        return `Soy el profesor ${this.getNombre()} y enseño ${this.materia}.`;
    }
}
exports.Profesor = Profesor;
class Estudiante extends encapsulamiento_1.Persona {
    constructor(nombre, edad, sexo, matricula) {
        super(nombre, edad, sexo);
        this.matricula = matricula;
    }
    getMatricula() {
        return this.matricula;
    }
    setMatricula(matricula) {
        this.matricula = matricula;
    }
    presentar() {
        return `Soy el estudiante ${this.getNombre()} y mi matrícula es ${this.matricula}.`;
    }
}
exports.Estudiante = Estudiante;
function presentarPersona(persona) {
    if (persona instanceof Profesor) {
        return persona.presentar();
    }
    else if (persona instanceof Estudiante) {
        return persona.presentar();
    }
    else {
        return `Soy ${persona.getNombre()} y tengo ${persona.getEdad()} años.`;
    }
}

File: tests/src/polimorfismo.py
from encapsulamiento import Persona, Planta
from herencia import Estudiante, Arbol

class Profesor(Persona):
    def __init__(self, nombre: str, edad: int, materia: str):
        super().__init__(nombre, edad)
        self.__materia = materia
    
    def get_materia(self) -> str:
        return self.__materia
    
    def set_materia(self, materia: str) -> None:
        self.__materia = materia
    
    def presentar(self) -> str:
        return f"Soy el profesor {self.get_nombre()} y enseño {self.__materia}."

class Estudiante(Persona):
    def __init__(self, nombre: str, edad: int, matricula: str):
        super().__init__(nombre, edad)
        self.__matricula = matricula

    def get_matricula(self) -> str:
        return self.__matricula

    def set_matricula(self, matricula: str) -> None:
        self.__matricula = matricula

    def presentar(self) -> str:
        return f"Soy el estudiante {self.get_nombre()} y mi matricula es {self.__matricula}."
    
class Hortaliza(Planta):
    def __init__(self, nombre: str, origen: str, clima: str, utilidad: str):
        super().__init__(nombre, origen, clima)
        self.__utilidad = utilidad

    def get_utilidad(self) -> str:
        return self.__utilidad

    def set_utilidad(self, utilidad: str) -> None:
        self.__utilidad = utilidad

    def exponer(self) -> str:
        return f"Soy una hortaliza, me llamo {self.get_nombre()} y mi utilidad es {self.__utilidad}."

class Arbol(Planta):
    def __init__(self, nombre: str, origen: str, clima: str, altura: int):
        super().__init__(nombre, origen, clima)
        self.__altura = altura

    def get_altura(self) -> int:
        return self.__altura

    def set_altura(self, altura: int) -> None:
        self.__altura = altura

    def exponer(self) -> str:
        return f"Soy un arbol, me llamo {self.get_nombre()} y mi altura es {self.__altura} m."


def exponer_planta(planta: Planta) -> str:
    return planta.exponer()

def presentar_persona(persona: Persona) -> str:
    return persona.presentar()